### PR TITLE
ENH: Add from_dummies

### PR DESCRIPTION
**Pull request by bashtage**
- [X] xref #227
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: pandas-stubs/core/reshape/encoding.pyi [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+from pandas._typing import (
+    ArrayLike,
+    Dtype,
+    HashableT,
+)
+
+def get_dummies(
+    data: ArrayLike | DataFrame | Series,
+    prefix: str | Iterable[str] | dict[str, str] | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/reshape/encoding.pyi:19]: For the `dict` case, docs say "prefix can be a dictionary mapping column names to prefixes", so maybe this should be `dict[Hashable, str]`

File: pandas-stubs/core/reshape/reshape.pyi [hunk 2]
```diff
@@ -1,16 +1,9 @@
...
-from pandas.core.frame import DataFrame
+from pandas import (
+    DataFrame,
+    Series,
+)
 
-def unstack(obj, level, fill_value=...): ...
-def stack(frame, level: int = ..., dropna: bool = ...): ...
+def unstack(obj: Series | DataFrame, level, fill_value: object | None = ...): ...
+def stack(frame: DataFrame, level: int = ..., dropna: bool = ...): ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/reshape/reshape.pyi:8]: I don't think that `stack` or `unstack` are public in `reshape`.  We have `DataFrame.stack()`, `DataFrame.unstack()`, etc. so I think these 2 can be deleted.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Some small things to address

File: pandas-stubs/core/reshape/encoding.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/core/reshape/encoding.pyi]: Fixed.

File: pandas-stubs/core/reshape/reshape.pyi (hunk 2 above)
Comment (bashtage) [reply pandas-stubs/core/reshape/reshape.pyi]: I removed the entire module.  CHecked other function, not in docs.

Comment (Dr-Irv) [reply pandas-stubs/core/reshape/reshape.pyi]: OK, good idea

File: pandas-stubs/core/reshape/encoding.pyi (hunk 1 above)
Comment (Dr-Irv) [reply pandas-stubs/core/reshape/encoding.pyi]: thanks

**Pull request merged**